### PR TITLE
Bundle htslib and bcftools

### PR DESCRIPTION
**Pull request by xbrianh**
The bundled samtools software includes [htslib](https://github.com/samtools/htslib) and [bcftools](https://github.com/samtools/bcftools), which have been modified to merge multisample, completely aligned VCFs without requiring index files.

### CODE REVIEW

File: terra_notebook_utils/samtools.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+bin_root = os.path.abspath(os.path.dirname(sys.executable))
+htsfile_path = os.path.join(bin_root, "htsfile")
+bcftools_path = os.path.join(bin_root, "bcftools")
+bcftools_available = os.path.isfile(bcftools_path)
+
+try:
+    p = _run([htsfile_path, "--version"], capture_output=True)
+    htsfile_available = True
+except (FileNotFoundError, subprocess.CalledProcessError):
```
Comment (DailyDreaming) [inline terra_notebook_utils/samtools.py:18]: Maybe something like:

```
except (FileNotFoundError, subprocess.CalledProcessError) as e:
    print(e)  # or traceback.print_exc()
    htsfile_available = False
```

Comment (xbrianh) [reply terra_notebook_utils/samtools.py:18]: An exception here indicates htlib failed to build during `setup.py`. Printing out the trace would not help diagnose that problem, and would therefore be confusing.

We could do something like:
```
except (FileNotFoundError, subprocess.CalledProcessError) as e:
    print("htsfile unavailable: htslib build failed during installation. try `pip install -v terra-notebook-utils` to diagnose the problem")
    htsfile_available = False
```

Comment (xbrianh) [reply terra_notebook_utils/samtools.py:18]: done.

**Comment by xbrianh** [comment]
Closed in favor of a separate pip installable package [xsamtools](https://github.com/xbrianh/xsamtools)

**Pull request closed without merging**